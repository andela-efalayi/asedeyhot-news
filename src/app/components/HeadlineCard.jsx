import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Card, CardActions, CardHeader,
  CardMedia, CardText } from 'material-ui/Card';
import RaisedButton from 'material-ui/RaisedButton';
import IconButton from 'material-ui/IconButton';
import FavoriteBorder from 'material-ui/svg-icons/action/favorite-border';
import Favorite from 'material-ui/svg-icons/action/favorite';

import DatabaseActions from '../actions/DatabaseActions';

class HeadlineCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: this.props.headline,
      favIcon: <FavoriteBorder />,
      tooltip: 'add to archive'
    };
    this.addToArchive = this.addToArchive.bind(this);
  }
  addToArchive = () => {
    DatabaseActions.addToArchive(this.state.value);
    this.setState({
      favIcon: <Favorite />,
      tooltip: 'added to archives'
    });
  };
  render() {
    return (
      <Card className="col-xs-6 col-sm-6 col-md-4 card withImage">
      <CardHeader title={this.props.headline.title} />
      <CardMedia >
        <img src={this.props.headline.urlToImage} />
      </CardMedia>
      <CardText className="cardDescription">
        <span>{this.props.headline.description}</span>
        <span className="readMore">
          <RaisedButton
          label="Read more"
          href={this.props.headline.url}
          target="_blank"
          secondary
        />
        </span>
      </CardText>
      <CardActions className="cardActions">
        <IconButton tooltip={this.state.tooltip} touch
        onClick={this.addToArchive} tooltipPosition="top-right">
        {this.state.favIcon}
        </IconButton>>
      </CardActions>
    </Card>
    );
  }
}

HeadlineCard.propTypes = {
  headline: PropTypes.object
};

HeadlineCard.defaultProps = {
  headline: null
};

export default HeadlineCard;
