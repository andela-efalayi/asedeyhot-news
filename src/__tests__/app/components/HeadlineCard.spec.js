import React from 'react';
import { shallow } from 'enzyme';
import HeadlineCard from '../../../app/components/HeadlineCard.jsx';

const demo = {
  title: 'Testing headline card',
  urlToImage: 'https://dummyimage.com/600x400/000000/0011ff',
  description: 'some description',
  url: 'http://www.aljazeera.com',
};

describe('HeadlineCard Component', () => {
  it('should render a HeadlineCard for a headline/article', () => {
    const wrapper = shallow(
        <HeadlineCard headline={demo}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
