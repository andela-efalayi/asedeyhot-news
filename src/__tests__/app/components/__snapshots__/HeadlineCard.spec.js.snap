// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadlineCard Component should render a HeadlineCard for a headline 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card
    className="col-xs-6 col-sm-6 col-md-4 card withImage"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
>
    <CardHeader
        avatar={null}
        title="Testing headline card"
    />
    <CardMedia>
        <img
            src="https://dummyimage.com/600x400/000000/0011ff"
        />
    </CardMedia>
    <CardText
        className="cardDescription"
    >
        <span>
            some description
        </span>
        <span
            className="readMore"
        >
            <RaisedButton
                disabled={false}
                fullWidth={false}
                href="http://www.aljazeera.com"
                label="Read more"
                labelPosition="after"
                primary={false}
                secondary={true}
                target="_blank"
            />
        </span>
    </CardText>
    <CardActions
        className="cardActions"
    >
        <IconButton
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            onClick={[Function]}
            tooltip="add to archive"
            tooltipPosition="top-right"
            touch={true}
        >
            <ActionFavoriteBorder />
        </IconButton>
        &gt;
    </CardActions>
</Card>,
  "nodes": Array [
    <Card
      className="col-xs-6 col-sm-6 col-md-4 card withImage"
      expandable={false}
      expanded={null}
      initiallyExpanded={false}
>
      <CardHeader
            avatar={null}
            title="Testing headline card"
      />
      <CardMedia>
            <img
                  src="https://dummyimage.com/600x400/000000/0011ff"
            />
      </CardMedia>
      <CardText
            className="cardDescription"
      >
            <span>
                  some description
            </span>
            <span
                  className="readMore"
            >
                  <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        href="http://www.aljazeera.com"
                        label="Read more"
                        labelPosition="after"
                        primary={false}
                        secondary={true}
                        target="_blank"
                  />
            </span>
      </CardText>
      <CardActions
            className="cardActions"
      >
            <IconButton
                  disableTouchRipple={false}
                  disabled={false}
                  iconStyle={Object {}}
                  onClick={[Function]}
                  tooltip="add to archive"
                  tooltipPosition="top-right"
                  touch={true}
            >
                  <ActionFavoriteBorder />
            </IconButton>
            &gt;
      </CardActions>
</Card>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <HeadlineCard
        googleId={null}
        headline={
                Object {
                        "description": "some description",
                        "title": "Testing headline card",
                        "url": "http://www.aljazeera.com",
                        "urlToImage": "https://dummyimage.com/600x400/000000/0011ff",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": HeadlineCard {
        "_reactInternalInstance": [Circular],
        "addToArchive": [Function],
        "context": Object {},
        "props": Object {
          "googleId": null,
          "headline": Object {
            "description": "some description",
            "title": "Testing headline card",
            "url": "http://www.aljazeera.com",
            "urlToImage": "https://dummyimage.com/600x400/000000/0011ff",
          },
        },
        "refs": Object {},
        "state": Object {
          "favIcon": <ActionFavoriteBorder />,
          "tooltip": "add to archive",
          "value": Object {
            "description": "some description",
            "title": "Testing headline card",
            "url": "http://www.aljazeera.com",
            "urlToImage": "https://dummyimage.com/600x400/000000/0011ff",
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card
          className="col-xs-6 col-sm-6 col-md-4 card withImage"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
>
          <CardHeader
                    avatar={null}
                    title="Testing headline card"
          />
          <CardMedia>
                    <img
                              src="https://dummyimage.com/600x400/000000/0011ff"
                    />
          </CardMedia>
          <CardText
                    className="cardDescription"
          >
                    <span>
                              some description
                    </span>
                    <span
                              className="readMore"
                    >
                              <RaisedButton
                                        disabled={false}
                                        fullWidth={false}
                                        href="http://www.aljazeera.com"
                                        label="Read more"
                                        labelPosition="after"
                                        primary={false}
                                        secondary={true}
                                        target="_blank"
                              />
                    </span>
          </CardText>
          <CardActions
                    className="cardActions"
          >
                    <IconButton
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="add to archive"
                              tooltipPosition="top-right"
                              touch={true}
                    >
                              <ActionFavoriteBorder />
                    </IconButton>
                    &gt;
          </CardActions>
</Card>,
        "_debugID": 2,
        "_renderedOutput": <Card
          className="col-xs-6 col-sm-6 col-md-4 card withImage"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
>
          <CardHeader
                    avatar={null}
                    title="Testing headline card"
          />
          <CardMedia>
                    <img
                              src="https://dummyimage.com/600x400/000000/0011ff"
                    />
          </CardMedia>
          <CardText
                    className="cardDescription"
          >
                    <span>
                              some description
                    </span>
                    <span
                              className="readMore"
                    >
                              <RaisedButton
                                        disabled={false}
                                        fullWidth={false}
                                        href="http://www.aljazeera.com"
                                        label="Read more"
                                        labelPosition="after"
                                        primary={false}
                                        secondary={true}
                                        target="_blank"
                              />
                    </span>
          </CardText>
          <CardActions
                    className="cardActions"
          >
                    <IconButton
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="add to archive"
                              tooltipPosition="top-right"
                              touch={true}
                    >
                              <ActionFavoriteBorder />
                    </IconButton>
                    &gt;
          </CardActions>
</Card>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HeadlineCard
    googleId={null}
    headline={
        Object {
            "description": "some description",
            "title": "Testing headline card",
            "url": "http://www.aljazeera.com",
            "urlToImage": "https://dummyimage.com/600x400/000000/0011ff",
          }
    }
/>,
}
`;
